pagerduty/resource_pagerduty_service_dependency.go

```diff
@@ -193,17 +197,17 @@ func resourcePagerDutyServiceDependencyDisassociate(d *schema.ResourceData, meta
 	input := pagerduty.ListServiceDependencies{
 		Relationships: r,
 	}
-	retryErr := resource.Retry(30*time.Second, func() *resource.RetryError {
+	retryErr = resource.Retry(5*time.Minute, func() *resource.RetryError {
 		if _, _, err = client.ServiceDependencies.DisassociateServiceDependencies(&input); err != nil {
-			if isErrCode(err, 404) {
+			if isErrCode(err, 404) || isErrCode(err, 429) {
 				return resource.RetryableError(err)
```

Comment:
@stmcallister I am a bit concerned the end users might have to wait for 5 min to find out a resource already exists or had already been deleted manually which are not that uncommon. 

One solution that seems fairly easy to implement at a glance would be to add a max timeout to the retryable error as an extra optional parameter

```suggestion
                        if isErrCode(err, 429) {
				return resource.RetryableError(err)
			} else if  isErrCode(err, 404) {
				return resource.RetryableErrorWithMaxTimeout(err, 15 * time.Second)
			}
```